pysatCDAAC/instruments/methods/general.py

```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
+"""Provides default routines for CDAAC instruments into pysat."""
+
+import os
+import requests
+import tarfile
+import tempfile
+import warnings
+
+import pysat
+
+
+def download(date_array, tag, inst_id, supported_tags=None,
+             data_path=None, sub_path=False, user=None, password=None):
+    """Download data from CDAAC https server.
+
+    Parameters
+    ----------
+    date_array : array-like
+        list of datetimes to download data for. The sequence of dates need not
+        be contiguous.
+    tag : str
+        Tag identifier used for particular dataset. This input is provided by
+        pysat. (default='')
+    inst_id : str
+        Instrument ID string identifier used for particular dataset. This input
+        is provided by pysat. (default='')
+    supported_tags : dict
+        dict of dicts. Keys are supported tag names for download. Value is
+        a dict with 'remote_dir', 'tar_name'. Intended to be pre-set with
+        functools.partial then assigned to new instrument code.
+        (default=None)
+    data_path : str
+        Path to directory to download data to. (default=None)
+    sub_path : bool
+        If True, break up data into further subdirectories based on date.
+        (default=False)
+    user : str or NoneType
+        User string input used for download. Provided by user and passed via
+        pysat. If an account is required for downloads this routine here must
+        error if user not supplied. (default=None)
+    password : str or NoneType
+        Password for data download. (default=None)
+
+    Note
+    ----
+    This routine is invoked by pysat and is not intended for direct use by
+    the end user.
+
+    """
+
+    # Set up temporary directory for tar files
+    temp_dir = tempfile.TemporaryDirectory()
+
+    if tag is None:
+        tag = ''
+    if inst_id is None:
+        inst_id = ''
+    try:
+        inst_dict = supported_tags[inst_id][tag]
+    except KeyError:
+        raise ValueError('inst_id / tag combo unknown.')
```

Comment:
I believe these checks are done in pysat and shouldn't be done here.  Also, tag and inst_id are str types, not also NoneType.